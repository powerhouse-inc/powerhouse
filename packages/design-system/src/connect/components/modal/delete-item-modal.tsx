import {
  ConfirmationModalProps,
  ConnectConfirmationModal,
} from "./confirmation-modal";

export interface ConnectDeleteItemModalProps
  extends Omit<ConfirmationModalProps, "onContinue" | "continueLabel"> {
  readonly onDelete: () => void;
  readonly deleteLabel: string;
}

export function ConnectDeleteItemModal(props: ConnectDeleteItemModalProps) {
  const { onDelete, deleteLabel, ...restProps } = props;

  return (
    <ConnectConfirmationModal
      {...restProps}
      containerProps={{ className: "w-[450px]" }}
      continueButtonProps={{
        className: "bg-red-900",
      }}
      continueLabel={deleteLabel}
      onContinue={onDelete}
    />
  );
}
