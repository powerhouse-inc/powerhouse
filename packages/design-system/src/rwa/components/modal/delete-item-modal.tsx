import { Icon, Modal } from "@/powerhouse";
import { DependentItemProps, tableLabels, TableName } from "@/rwa";
import { ComponentPropsWithoutRef, memo, useCallback } from "react";

export type RWADeleteItemModalProps = ComponentPropsWithoutRef<typeof Modal> & {
  readonly tableName: TableName;
  readonly dependentItemProps: DependentItemProps;
};

export function _RWADeleteItemModal(props: RWADeleteItemModalProps) {
  const { tableName, dependentItemProps, onOpenChange, open } = props;

  const tableLabel = tableLabels[tableName].toLowerCase();

  const handleCancel = useCallback(() => {
    onOpenChange?.(false);
  }, [onOpenChange]);
  return (
    <Modal
      contentProps={{
        className: "rounded-3xl",
      }}
      onOpenChange={onOpenChange}
      open={open}
      overlayProps={{
        className: "top-10",
      }}
    >
      <div className="w-[400px] p-6 text-slate-300">
        <div className="border-b border-slate-50 pb-2 text-2xl font-bold text-gray-800">
          Cannot delete {tableName}
        </div>
        <div className="my-6 flex gap-2 rounded-md bg-orange-100 p-4 text-orange-800">
          <div>
            <Icon className="mt-1 text-orange-800" name="Error" />
          </div>
          <div>
            Warning! Cannot delete this {tableLabel} because there are items
            that depend on it. Please change or delete those first.
          </div>
        </div>
        <div className="my-6 rounded-md bg-slate-50 p-4 text-slate-200">
          {dependentItemProps.map(({ dependentTableName, dependentItems }) => (
            <div key={dependentTableName}>
              <div className="mb-0.5 font-semibold" key={1}>
                {`${tableLabels[dependentTableName]}s: `}
              </div>
              {dependentItems.map(({ id, label }) => (
                <div key={id}>{label}</div>
              ))}
            </div>
          ))}
        </div>
        <div className="mt-8 flex justify-between gap-3">
          <button
            className="min-h-12 min-w-36 flex-1 rounded-xl bg-gray-800 px-6 py-3 text-base font-semibold text-gray-50 outline-none transition-all hover:scale-105 active:opacity-75"
            onClick={handleCancel}
          >
            Back
          </button>
        </div>
      </div>
    </Modal>
  );
}

export const RWADeleteItemModal = memo(_RWADeleteItemModal);
