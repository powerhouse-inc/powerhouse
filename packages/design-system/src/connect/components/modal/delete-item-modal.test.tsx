import { fireEvent, render, screen } from "@testing-library/react";
import { ConnectDeleteItemModal } from "./delete-item-modal";

describe("Modal Component", () => {
  it("should match snapshot", () => {
    render(
      <ConnectDeleteItemModal
        body="Are you sure you want to delete this folder? All files and subfolders within it will be removed."
        cancelLabel="Cancel"
        data-testid="delete-modal"
        deleteLabel="Delete"
        header="Delete “Ecosystem Actors” folder?"
        onCancel={() => {}}
        onDelete={() => {}}
        open
      />,
    );

    const modalComponent = screen.getByTestId("delete-modal");

    expect(modalComponent).toMatchSnapshot();
  });

  it("should render modal content", () => {
    const headerText = "Delete “Ecosystem Actors” folder?";
    const bodyText =
      "Are you sure you want to delete this folder? All files and subfolders within it will be removed.";
    const cancelLabel = "Cancel";
    const deleteLabel = "Delete";

    render(
      <ConnectDeleteItemModal
        body={bodyText}
        cancelLabel={cancelLabel}
        data-testid="delete-modal"
        deleteLabel={deleteLabel}
        header={headerText}
        onCancel={() => {}}
        onDelete={() => {}}
        open
      />,
    );

    const header = screen.getByText(headerText);
    const body = screen.getByText(bodyText);
    const cancelButton = screen.getByText(cancelLabel);
    const deleteButton = screen.getByText(deleteLabel);

    expect(header).toBeInTheDocument();
    expect(body).toBeInTheDocument();
    expect(cancelButton).toBeInTheDocument();
    expect(deleteButton).toBeInTheDocument();
  });

  it("should trigger onClose callback when cancel is clicked", () => {
    const onClose = vi.fn();
    const cancelLabel = "Cancel";

    render(
      <ConnectDeleteItemModal
        body="Are you sure you want to delete this folder? All files and subfolders within it will be removed."
        cancelLabel={cancelLabel}
        data-testid="delete-modal"
        deleteLabel="Delete"
        header="Delete “Ecosystem Actors” folder?"
        onCancel={onClose}
        onDelete={() => {}}
        open
      />,
    );

    fireEvent.click(screen.getByText(cancelLabel));

    expect(onClose).toHaveBeenCalled();
  });

  it("should trigger onDelete callback when CTA is clicked", () => {
    const onDelete = vi.fn();
    const deleteLabel = "Delete";

    render(
      <ConnectDeleteItemModal
        body="Are you sure you want to delete this folder? All files and subfolders within it will be removed."
        cancelLabel="Cancel"
        data-testid="delete-modal"
        deleteLabel={deleteLabel}
        header="Delete “Ecosystem Actors” folder?"
        onCancel={() => {}}
        onDelete={onDelete}
        open
      />,
    );

    fireEvent.click(screen.getByText(deleteLabel));

    expect(onDelete).toHaveBeenCalled();
  });
});
