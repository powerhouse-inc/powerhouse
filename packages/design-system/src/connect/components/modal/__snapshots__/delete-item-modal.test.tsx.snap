// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal Component > should match snapshot 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="bg-white data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in rounded-3xl"
  data-state="open"
  data-testid="delete-modal"
  id="radix-:r0:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <span
    style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
  >
    <h2
      id="radix-:r1:"
    >
      Modal
    </h2>
    <p
      id="radix-:r2:"
    >
      Modal
    </p>
  </span>
  <div
    class="p-6 text-slate-300 w-[450px]"
  >
    <div
      class="border-b border-slate-50 pb-2 text-2xl font-bold text-gray-800"
    >
      Delete “Ecosystem Actors” folder?
    </div>
    <div
      class="my-6 rounded-md bg-slate-50 p-4 text-center text-slate-200"
    >
      Are you sure you want to delete this folder? All files and subfolders within it will be removed.
    </div>
    <div
      class="mt-8 flex justify-between gap-3"
    >
      <button
        class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all flex-1 bg-slate-50 text-slate-800"
      >
        Cancel
      </button>
      <button
        class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all flex-1 text-gray-50 bg-red-900"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
